lgc/patch/PatchResourceCollect.cpp

```diff
@@ -2763,13 +2763,22 @@ void PatchResourceCollect::scalarizeForInOutPacking(Module *module) {
   SmallVector<CallInst *, 4> outputCalls;
   SmallVector<CallInst *, 4> inputCalls;
   for (Function &func : *module) {
-    if (func.getName().startswith(lgcName::InputImportGeneric) ||
-        func.getName().startswith(lgcName::InputImportInterpolant)) {
+    if (!m_pipelineState->canPackInOut())
+      break;
+    const bool isInterpolant = func.getName().startswith(lgcName::InputImportInterpolant);
+    if (func.getName().startswith(lgcName::InputImportGeneric) || isInterpolant) {
       // This is a generic (possibly interpolated) input. Find its uses in FS (VS-FS, TES-FS) or TCS.
       for (User *user : func.users()) {
         auto call = cast<CallInst>(user);
         ShaderStage shaderStage = m_pipelineShaders->getShaderStage(call->getFunction());
         if (shaderStage == ShaderStageFragment || shaderStage == ShaderStageTessControl) {
+          // This is a workaround to disable pack for the pipeline if there exists dynamic component index
+          // TODO: Do partial packing except calls with dynamic component index in future change
+          const unsigned elemArgIdx = shaderStage == ShaderStageFragment && !isInterpolant ? 1 : 2;
```

Comment:
nit: Can you add parentheses around the condition part of the ternary operator? Or alternatively, hoist it into a boolean variable.

Reply:
Sure. Add parentheses.